FortnitePorting/Plugins/Blender/import_task.py

```diff
@@ -563,14 +591,153 @@ def get_meta(search_props):
         match mesh_type:
             case "Body":
                 meta = get_meta(["SkinColor"])
-            case "Head":
-                meta = get_meta(["MorphNames", "HatType", "PoseData"])
+            case "Head" | "Face":
+                meta = get_meta(["MorphNames", "HatType", "PoseData", "ReferencePose"])
+                is_face = mesh_type == "Face"
 
                 shape_keys = imported_mesh.data.shape_keys
                 if (morph_name := meta.get("MorphNames").get(meta.get("HatType"))) and shape_keys is not None:
                     for key in shape_keys.key_blocks:
                         if key.name.casefold() == morph_name.casefold():
                             key.value = 1.0
+
+                if (pose_data := meta.get("PoseData")):
+                    original_mode = bpy.context.active_object.mode
+                    try:
+                        bpy.ops.object.mode_set(mode='OBJECT')
+
+                        # Find the ARMATURE modifier
+                        armature_modifiers = [mod for mod in imported_mesh.modifiers if mod.type == "ARMATURE"]
+                        assert len(armature_modifiers) == 1, "expected at least one armature modifier"
```

Comment:
How do we feel about my use of assertions? Should I change these to formatted exceptions?

Reply:
tbh i've never really used assertions in python so i cant really comment on that
it might be a better design choice to let the export carry out without pose stuff if it doesnt work? bc right now there's no way to toggle pose asset loading so it could break entire assets which would be annoying for people who need the skin but arent using em

Reply:
Added some slightly better error handling w.r.t resetting pose bones on failed pose asset import. There may be partially incomplete keys, but otherwise I think it's an improvement. Let me know if you'd like to see any additional changes here.